src/main/kotlin/systems/ajax/englishstudytelegrambot/config/LogMethodsByRequiredAnnotationsBeanPostProcessor.kt

```diff
@@ -0,0 +1,51 @@
+package systems.ajax.englishstudytelegrambot.config
+
+import org.aopalliance.intercept.MethodInterceptor
+import org.aopalliance.intercept.MethodInvocation
+import org.slf4j.LoggerFactory
+import org.springframework.aop.framework.ProxyFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import systems.ajax.englishstudytelegrambot.annotation.LogMethodsByRequiredAnnotations
+import java.lang.reflect.Method
+import kotlin.reflect.KClass
+
+@Component
+class LogMethodsByRequiredAnnotationsBeanPostProcessor : BeanPostProcessor {
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        if (bean.javaClass.isAnnotationPresent(LogMethodsByRequiredAnnotations::class.java)) {
+            LoggingInterceptor.requiredAnnotations =
+                bean.javaClass.getAnnotation(LogMethodsByRequiredAnnotations::class.java).requiredAnnotations
+            val proxyFactory = ProxyFactory(bean);
+            proxyFactory.addAdvice(LoggingInterceptor);
+            return proxyFactory.getProxy();
+        }
+        return bean;
+    }
```

Comment:
will it work with multiple bean post processors?

Reply:
you mean if i add new bpp it will not work? I haven't test the scenario, have to try

Reply:
but under the hoot in spring the are a lot bpps, so i think it has to work